cmd/editorconfig-checker/main.go

```diff
@@ -2,57 +2,70 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
+	"io/fs"
 	"os"
 	"strings"
 
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/config"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/error"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files"
-	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/logger"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/outputformat"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/utils"
 	"github.com/editorconfig-checker/editorconfig-checker/v3/pkg/validation"
 )
 
-// version is used vor the help
+// version is used for the help and to verify against the version stored in the config file
 // version is dynamically set at compiletime
-var version string
+var version string = "v3.0.3"
```

Comment:
I would leave coming up with a better strategy up to #385 - here I just rebased my change to work nicely with #385 ( I introduced a magic string "development" as the version content and checked for that in a piece later, but this will no longer be needed after #385, maybe in two days time)